### ISSUE TITLE
psycopg2.ProgrammingError: ON CONFLICT DO UPDATE command cannot affect row a second time

### ISSUE THREAD

**Issue by de-code** [elifesciences/data-pipeline-sql-prototype#117]
After testing fix for #113, another obscure error :-( 

### 1. In _csv-generator_ (or using Docker):

```bash
pipenv run python -m csv_generator.process_xml_zip \
  --source-zip /path/to/ejp_eLife_2018_01_01_00_00_00_2018_04_17_23_59_59.zip\
  --output-dir ../output

pipenv run python -m csv_generator.process_xml_zip \
  --source-zip /path/to/ejp_eLife_2018_01_01_02_01_39_2018_01_02_02_01_39.zip\
  --output-dir ../output
```

### 2. In _db-manager_ (or using Docker):

Clean database:

```bash
pipenv run python -m db_manager teardown &&
pipenv run python -m db_manager create
```

### 3. In _db-manager_ (or using Docker):

```bash
pipenv run python -m db_manager import-data --source-dir ../output/
```

This time it fails with:

```
INFO:db_manager.processing:staging: db_manager.dimManuscriptVersionHistory {'file_path': '../output/1523954246_stages.csv'}
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/__main__.py", line 88, in <module>
    main()
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/__main__.py", line 76, in main
    is_batch_mode=args.batch
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/processing.py", line 151, in process_source_dir
    process_filenames(connection, filenames, is_batch_mode=is_batch_mode)
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/processing.py", line 144, in process_filenames
    connection, staging_instructions, is_batch_mode=is_batch_mode
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/processing.py", line 135, in process_staging_instructions_group
    staging_module.applyChanges(connection, source=None)
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/dimPerson.py", line 211, in applyChanges
    dimPersonRole.applyChanges(conn, 'dimPerson')
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/dimPersonRole.py", line 378, in applyChanges
    _applyChanges(conn)
  File "/home/deuser/_git_/elife/data-pipeline/data-pipeline/db-manager/db_manager/dimPersonRole.py", line 433, in _applyChanges
    """
psycopg2.ProgrammingError: ON CONFLICT DO UPDATE command cannot affect row a second time
HINT:  Ensure that no rows proposed for insertion within the same command have duplicate constrained values.
```

## Expected behaviour:

It should not fail to import any file, except due to validation errors within _csv-generator_.

## Potential cause

Condition near ON CONFLICT DO UPDATE not specific enough?

**Comment by de-code** [comment]
I've added a `person-role-tests` branch but haven't reproduced the error with the tests yet.

**Comment by lsh-0** [comment]
I can't replicate this, but I've just noticed you're not doing what is being done in #113, which is inserting the same data multiple times, you're inserting two completely different sets of data. Let me try again

**Comment by lsh-0** [comment]
ok, I can replicate this
